src/internal.rs

```diff
@@ -412,7 +427,9 @@ pub struct Rav1dFrameContext_frame_thread {
 
     /// Indexed using `(t.by >> 1) * (f.b4_stride >> 1) + (t.bx >> 1)`.
     /// Inner indices are `[3 plane][8 idx]`.
-    pub pal: AlignedVec64<[[u16; 8]; 3]>,
+    /// Allocated as a flat array. `pal_as_slice` and `pal_as_slice_mut` should be
+    /// used to access elements of type `[[BitDepth::Pixel; 8]; 3]`
+    pub pal: AlignedVec64<u8>,
```

Comment:
This might run into a borrowck issue later, as `.pal_as_slice()` borrows the whole `&Rav1dFrameContext_frame_thread` instead `.pal` just borrowing the field.  We might want to newtype `pal` and have a `.as_slice()` method on it instead.  That way we can do `.pal.as_slice()`, which will allow borrowck to see we're only borrowing the one field.